daemon/cmd/daemon.go

```diff
@@ -490,3 +390,77 @@ func configureDaemon(ctx context.Context, cleaner *daemonCleanup, params daemonP
 
 	return nil
 }
+
+func unloadDNSPolicies(params daemonParams) {
+	if option.Config.DNSPolicyUnloadOnShutdown {
+		// Stop k8s watchers
+		params.Logger.Info("Stopping k8s watcher")
+		params.K8sWatcher.StopWatcher()
```

Comment:
tbh: no clue why we stop the k8s watcher only when the feature to unload dns policies on shutdown is enabled (disabled by default) 😕 

But i'd like to keep it this way - at least in this PR.